hack/lsplatform/lsplatform.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/k8stopologyawareschedwg/deployer/pkg/deployer/platform"
+	"github.com/k8stopologyawareschedwg/deployer/pkg/deployer/platform/detect"
+)
+
+func main() {
+	platformName := ""
+	flag.StringVar(&platformName, "is-platform", "", "check if the detected platform matches the given one")
+	flag.Parse()
+
+	clusterPlatform, err := detect.Detect()
+	if err != nil {
+		log.Fatalf("error detecting the platform: %v", err)
+	}
+
+	if platformName != "" {
+		ret := 1
+		userPlatform, ok := platform.FromString(platformName)
+		if !ok {
+			log.Fatalf("error parsing the user platform: %q", platformName)
+		}
+
+		if userPlatform == clusterPlatform {
+			ret = 0
```

Comment:
shouldn't it return 1 if the platforms are matching?

Reply:
we're following the bash (shell) conventions here.
I conditionals, the unix shell interprets 0 as success, because 0 means success, meaning the command executed without errors: https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_01.html

Reply:
ok good to know that, thanks for the information!